src/main/java/org/opentripplanner/graph_builder/module/interlining/InterlineProcessor.java

```diff
@@ -0,0 +1,184 @@
+import org.opentripplanner.graph_builder.DataImportIssueStore;
+import org.opentripplanner.graph_builder.issues.InterliningTeleport;
+import org.opentripplanner.gtfs.mapping.StaySeatedNotAllowed;
+import org.opentripplanner.model.Timetable;
+import org.opentripplanner.model.TripPattern;
+import org.opentripplanner.model.transfer.ConstrainedTransfer;
+import org.opentripplanner.model.transfer.TransferConstraint;
+import org.opentripplanner.model.transfer.TransferPriority;
+import org.opentripplanner.model.transfer.TransferService;
+import org.opentripplanner.model.transfer.TripTransferPoint;
+import org.opentripplanner.routing.trippattern.TripTimes;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InterlineProcessor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(InterlineProcessor.class);
+  private final TransferService transferService;
+  private final int maxInterlineDistance;
+  private final DataImportIssueStore issueStore;
+  private List<StaySeatedNotAllowed> staySeatedNotAllowed;
+
+  public InterlineProcessor(
+    TransferService transferService,
+    List<StaySeatedNotAllowed> staySeatedNotAllowed,
+    int maxInterlineDistance,
+    DataImportIssueStore issueStore
+  ) {
+    this.transferService = transferService;
+    this.staySeatedNotAllowed = staySeatedNotAllowed;
+    this.maxInterlineDistance = maxInterlineDistance > 0 ? maxInterlineDistance : 200;
+    this.issueStore = issueStore;
+  }
+
+  public List<ConstrainedTransfer> run(Collection<TripPattern> tripPatterns) {
+    var interlinedTrips = this.getInterlinedTrips(tripPatterns);
+    var transfers = interlinedTrips
+      .entries()
+      .stream()
+      .filter(this::staySeatedNotForbidden)
+      .map(p -> {
+        var constraint = TransferConstraint.create();
+        constraint.staySeated();
+        constraint.priority(TransferPriority.ALLOWED);
+
+        var fromTrip = p.getValue().first;
+        var toTrip = p.getValue().second;
+
+        var from = new TripTransferPoint(fromTrip, p.getKey().first.numberOfStops() - 1);
+        var to = new TripTransferPoint(toTrip, 0);
+
+        LOG.debug(
+          "Creating stay-seated transfer from trip {} (route {}) to trip {} (route {})",
+          fromTrip.getId(),
+          fromTrip.getRoute().getId(),
+          toTrip.getId(),
+          toTrip.getRoute().getId()
+        );
+
+        return new ConstrainedTransfer(null, from, to, constraint.build());
+      })
+      .toList();
+
+    if (!transfers.isEmpty()) {
+      LOG.info(
+        "Found {} pairs of trips for which stay-seated (interlined) transfers were created",
+        interlinedTrips.keySet().size()
+      );
+
+      transferService.addAll(transfers);
+    }
+    return transfers;
+  }
+
+  private boolean staySeatedNotForbidden(Map.Entry<P2<TripPattern>, P2<Trip>> p) {
```

Comment:
Couldn't this be just named `staySeatedAllowed`?

Reply:
Done.